contracts/transfer/src/state.rs

```diff
@@ -597,17 +597,16 @@ impl TransferState {
             None => panic!("Account has no funds"),
         }
 
-        // if there is a value carried by the transaction but no key specified
-        // in the `receiver` field, we just give the value back to `sender`.
+        // add the value to the receiver account
         if moonlight_tx.value() > 0 {
-            let key = match moonlight_tx.receiver() {
-                Some(to) => to.to_raw_bytes(),
-                None => from_bytes,
-            };
```

Comment:
Actually this code was more performant, since it avoid to perform a redundant "to_raw_bytes" in case of No receiver specified

Reply:
You are right! Although we would only end up in this if statement when someone makes a transfer to one-self which is basically just a waste of gas so shouldn't happen that often. But still good catch! Fixed